libs/core/algorithms/tests/unit/datapar_algorithms/adjacentdifference_datapar.cpp

```diff
@@ -0,0 +1,96 @@
+//  Copyright (c) 2021 Srinivas Yadav
+//  Copyright (c) 2015 Daniel Bourgeois
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/include/datapar.hpp>
+#include <hpx/local/init.hpp>
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "../algorithms/adjacentdifference_tests.hpp"
+
+void adjacent_difference_test()
+{
+    using namespace hpx::execution;
+    test_adjacent_difference(simd);
+    test_adjacent_difference(par_simd);
+
+    test_adjacent_difference_async(simd(task));
+    test_adjacent_difference_async(par_simd(task));
+}
+
+template <typename IteratorTag>
+void test_adjacent_difference_exception()
+{
+    using namespace hpx::execution;
+    test_adjacent_difference_exception(simd, IteratorTag());
+    test_adjacent_difference_exception(par_simd, IteratorTag());
+
+    test_adjacent_difference_exception_async(simd(task), IteratorTag());
+    test_adjacent_difference_exception_async(par_simd(task), IteratorTag());
+}
+
+void adjacent_difference_exception_test()
+{
+    test_adjacent_difference_exception<std::random_access_iterator_tag>();
+    // test_adjacent_difference_exception<std::forward_iterator_tag>();
```

Comment:
Is it still supposed to be commented?

Reply:
no it was by mistake. I changed it locally. I will update it soon! Thanks :)